### ISSUE TITLE
Error using rrd4j 3.3 with Java 8

### ISSUE THREAD

**Issue by bplump** [rrd4j/rrd4j#119]
Using rrd4j 3.3 and Java 8, when calling RrdDbPool.requestRrdDb() gives the error below. Running the same code with Java 11 does not have this problem.

```
java.lang.NoSuchMethodError: java.nio.ByteBuffer.position(I)Ljava/nio/ByteBuffer;

	at org.rrd4j.core.RrdBackend.writeString(RrdBackend.java:251)
	at org.rrd4j.core.RrdBackend.writeString(RrdBackend.java:245)
	at org.rrd4j.core.RrdPrimitive.writeString(RrdPrimitive.java:91)
	at org.rrd4j.core.RrdString.set(RrdString.java:23)
	at org.rrd4j.core.Header.<init>(Header.java:53)
	at org.rrd4j.core.RrdDb.<init>(RrdDb.java:150)
	at org.rrd4j.core.RrdDb.<init>(RrdDb.java:101)
	at org.rrd4j.core.RrdDbPool.requestRrdDb(RrdDbPool.java:355)
	[...]
```

**Comment by bplump** [comment]
Whilst waiting for the full release, we built [a fork](https://github.com/atlassian/rrd4j) from commit ee0464e that does not exhibit this problem. It works under both Java 8 and Java 11.

I'm trying to figure out why there's a difference between that version and the officially-released version (there were a few changes made before the release).

The maven group/artifact of our fork are the same but the verison is `3.3-atlassian-rc1`. The artifact is stored on our public repository:
```
<repository>
    <id>atlassian-3rdparty</id>
    <url>https://packages.atlassian.com/maven/3rdparty</url>
</repository>
```

**Comment by fbacchella** [comment]
It's explained here: https://github.com/openhab/openhab1-addons/issues/5663#issuecomment-425674686

I tried the proposed patches (casting to Buffer, see https://github.com/apache/felix/pull/114 or many others) but without any success.

**Comment by bplump** [comment]
That explains it. I built our fork with Java 8.

Thanks for being so quick to reply.

**Comment by fbacchella** [comment]
it's resolved in version 3.3.1 without dirty trick in the source code.

It uses java 11 compiler but using the 1.7 runtime.
